### ISSUE TITLE
vtc 'shell': processes may not be terminated if the test fails

### ISSUE THREAD

**Issue by slimhazard** [varnishcache/varnish-cache#2469]
## Expected Behavior
Any process started with the ``varnish`` or ``shell`` commands in a VTC script should be terminated when the test ends, regardless of whether the test passed or failed.

## Current Behavior
Under certain circumstances, processes started with ``varnish`` or ``shell`` may continue to run after varnishtest exits if the test fails.

This apparently does not always happen, but I it does for the [sample VTC at this gist](https://gist.github.com/slimhazard/8abea31322eaf5a4e99d905d20d522ba). This had been test ``u00006.vtc`` in master, and we had a problem getting ``varnishlog -i ''`` (-i option with an empty string as argument) to fail with the expected output on SunOS. That test has now been removed in master, but in the attached example, the test fails on every platform (because the expected output string is nonsense).

When the test shown at the gist runs (on my Debian laptop as well), it fails, and then ``varnishd`` and ``varnishlog`` processes continue to run after varnishtest stops. They remain in the process table until they are killed.

## Steps to Reproduce
1. Run the sample test at [gist](https://gist.github.com/slimhazard/8abea31322eaf5a4e99d905d20d522ba) (``varnishtest process.vtc``).
2. After varnishtest exits, inspect the process table (``ps -ef | grep varnish``).

On Debian and SunOS, I then see two varnishd processes (management and child) and one varnishlog process. These continue to run until they are kill(1)ed.

The varnishd processes are started by the ``varnish`` command in the VTC test. The varnishlog process that does not terminate is the one started by the first ``shell`` command:
```
shell {
	exec varnishlog -n ${v1_name} -D -P ${tmpdir}/vlog.pid \
	    -w ${tmpdir}/vlog.bin
}
```

## Context
This had severe consequences for the SunOS host running VTEST -- eventually the process table was full, and we had to reboot the host.

Generally, anyone running varnishtest is at risk of similarly filling up the process table with idle processes.

## Your Environment
* Version used: current master
* Operating System and version: Debian stretch and SunOS 5.11
```
$ cat /etc/debian_version 
9.2
$ uname -a
Linux xxx 4.9.0-4-amd64 #1 SMP Debian 4.9.51-1 (2017-09-28) x86_64 GNU/Linux
```
```
$ uname -a
SunOS xxx 5.11 joyent_20160317T000621Z i86pc i386 i86pc
```

**Comment by bsdphk** [comment]
This is sort of an old problem, which we have mostly managed to mitigate during test-writing.

Other test-frameworks setup a process-group to contain all children, grandchildren etc. so that the entire process-tree can be nuked from orbit at the end (subject of course to nohup(1), daemon(3) etc.)

But at the very least, we should actively nuke the processes we knew we started.

**Comment by Dridi** [comment]
Should be solved by c2e16df, let's reopen if it still occurs.

**Comment by hermunn** [comment]
Since this is in varnishtest I remove the `backport-review` label. If anyone needs this in 4.1, update this ticket.
